packages/search/src/search.tsx

```diff
@@ -0,0 +1,424 @@
+			return;
+		}
+
+		const setValue = this.searchInput.current.value ?? '';
+		if ( setValue ) {
+			// Firefox needs clear or won't move cursor to end
+			this.searchInput.current.value = '';
+			this.searchInput.current.value = setValue;
+		}
+	};
+
+	handleSubmit = ( event: FormEvent ) => {
+		event.preventDefault();
+	};
+
+	render() {
+		const searchValue = this.state.keyword;
+		const placeholder = this.props.placeholder || __( 'Search…' );
+		const inputLabel = this.props.inputLabel;
+		const isOpenUnpinnedOrQueried = this.state.isOpen || ! this.props.pinned || searchValue;
+
+		const autocorrectProps = this.props.disableAutocorrect && {
+			autoComplete: 'off',
+			autoCorrect: 'off',
+			spellCheck: 'false' as const,
+		};
+
+		const searchClass = classNames( 'search', this.props.className, this.props.dir, {
+			'is-expanded-to-container': this.props.fitsContainer,
+			'is-open': isOpenUnpinnedOrQueried,
+			'is-searching': this.props.searching,
+			'is-compact': this.props.compact,
+			'has-focus': this.state.hasFocus,
+			'has-open-icon': ! this.props.hideOpenIcon,
+		} );
+
+		const fadeDivClass = classNames( 'search__input-fade', this.props.dir );
+		const inputClass = classNames( 'search__input', this.props.dir );
+
+		return (
+			<div dir={ this.props.dir } className={ searchClass } role="search">
+				<Spinner />
+				{ this.renderOpenIcon() }
+				<div className={ fadeDivClass }>
+					<form action="." onSubmit={ this.handleSubmit }>
+						<input
+							type="search"
+							id={ 'search-component-' + this.instanceId }
+							autoFocus={ this.props.autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
+							aria-describedby={ this.props.describedBy }
+							aria-label={ inputLabel ? inputLabel : __( 'Search' ) }
+							aria-hidden={ ! isOpenUnpinnedOrQueried }
+							className={ inputClass }
+							placeholder={ placeholder }
+							role="searchbox"
+							value={ searchValue }
+							ref={ this.searchInput }
+							onChange={ this.onChange }
+							onKeyUp={ this.keyUp }
+							onKeyDown={ this.keyDown }
+							onMouseUp={ this.props.onClick }
+							onFocus={ this.onFocus }
+							onBlur={ this.onBlur }
+							disabled={ this.props.disabled }
+							autoCapitalize="none"
+							dir={ this.props.dir }
+							maxLength={ this.props.maxLength }
+							minLength={ this.props.minLength }
+							{ ...autocorrectProps }
+						/>
+					</form>
+					{ this.renderStylingDiv() }
+				</div>
+				{ this.closeButton() }
+			</div>
+		);
+	}
+
+	renderOpenIcon() {
+		const enableOpenIcon = this.props.pinned && ! this.state.isOpen;
+
+		return (
+			<Button
+				className="search__icon-navigation"
+				ref={ this.openIcon }
+				onClick={ enableOpenIcon ? this.openSearch : this.focus }
+				tabIndex={ enableOpenIcon ? 0 : undefined }
+				onKeyDown={ enableOpenIcon ? this.openListener : undefined }
+				aria-controls={ 'search-component-' + this.instanceId }
+				aria-label={ __( 'Open Search' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'Open Search', { context: 'button label'} )` *ES Score: 14*